plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt

```diff
@@ -163,6 +163,42 @@ class CameraAnimationsPluginImpl : CameraAnimationsPlugin {
     cameraAnimationsFactory = CameraAnimatorsFactory(mapDelegateProvider)
   }
 
+  /**
+   * Called whenever camera position changes.
+   * Could be invoked from any thread when map starts rendering.
+   *
+   * @param lat latitude
+   * @param lon longitude
+   * @param zoom zoom
+   * @param pitch pitch in degrees
+   * @param bearing bearing in degrees
+   * @param padding padding ordered as [left, top, right, bottom]
+   */
+  override fun onCameraMove(
```

Comment:
Question: since the comment indicates this could be called on any thread, does it mean we can be called from multiple threads concurrently? If so, then the internal state could be corrupted since we are saving values to `this`

Reply:
Indeed. The doc about this states that it might happen when `setCamera` is called directly after `mapView.onStart`, https://github.com/mapbox/mapbox-maps-android/issues/308. 
Not sure it's still possible since mapView was pushed to work only with UiThread since. Trying locally shows main thread is used even with the described calls sequence.

Reply:
Suppose gl-native somehow has changed the implementation of observer, I'm seeing that `MapEvents.CAMERA_CHANGED` event is delivered on the MainThread even setting camera directly after `onStart.` 
@alexshalamov @pozdnyakov do you folks remember if events are _always_ delivered on the main thread? We could drop the unneeded thread switches then.

Reply:
Talked offline with @pozdnyakov - indeed camera events originating in Observer API will arrive on Main thread. Checked it as well, works as expected.
Removed the comment about AnyThread here and in CompassViewPlugin as well.